pkg/sql/importer/read_import_avro.go

```diff
@@ -28,29 +29,47 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
 	"github.com/cockroachdb/cockroach/pkg/sql/types"
 	"github.com/cockroachdb/cockroach/pkg/util/ctxgroup"
+	"github.com/cockroachdb/cockroach/pkg/util/timeofday"
+	"github.com/cockroachdb/errors"
 	"github.com/linkedin/goavro/v2"
 )
 
-// nativeToDatum converts go native type (interface{} as
-// returned by goavro library) to the datum of appropriate type.
+// mapDateTime maps the time.Time object returned by goAvro to the proper CRL datum type.
+func mapDateTime(x interface{}, targetT *types.T) (tree.Datum, error) {
```

Comment:
[nit] Maybe nativeTimeToDatum to match the other function names?

Also, maybe the caller should do the type assertion and we should just take a time.time as an argument?